File: serverApp/src/main/java/com/server/dto/UserInfoDTO.java
package com.server.dto;

import com.server.enums.UserRoles;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.sql.Date;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class UserInfoDTO {
    private String email;

    private String login;

    private Date birthday;

    private UserRoles role;

    private Boolean isBlocked;
}

File: guide_app/lib/features/profile/dto/user_info_dto.dart
import 'package:json_annotation/json_annotation.dart';

part 'user_info_dto.g.dart';

/// User info dto.
/// For code generation run  flutter pub run build_runner build --delete-conflicting-outputs.
@JsonSerializable()
class UserInfoDto {
  UserInfoDto(this.email, this.login, this.birthday, this.role, this.isBlocked);
  final String email;
  final String login;
  final DateTime birthday;
  final String role;
  final bool isBlocked;

  factory UserInfoDto.fromJson(Map<String, dynamic> json) =>
      _$UserInfoDtoFromJson(json);
}
